nzedb/processing/Videos.php

```diff
@@ -84,15 +84,283 @@ protected function getSiteIDFromVideoID($siteColumn, $videoID)
+				return $res['id'];
+			}
+			$pieces = explode(' ', $title2);
+			$title4 = '%';
+			foreach ($pieces as $piece) {
+				$title4 .= str_replace(["'", "!"], "", $piece) . '%';
+			}
+			$res = $this->getByTitleLikeQuery($title4, $type, $source);
+			if (isset($res['id'])) {
+				return $res['id'];
+			}
+		}
+
+		// Some words are spelled correctly 2 ways
+		// example theatre and theater
+		$title3 = str_replace('er', 're', $title);
+		if ($title != $title3) {
+			$res = $this->getByTitleQuery($title3, $type, $source);
+			if (isset($res['id'])) {
+				return $res['id'];
+			}
+			$pieces = explode(' ', $title3);
+			$title4 = '%';
+			foreach ($pieces as $piece) {
+				$title4 .= str_replace(["'", "!"], "", $piece) . '%';
+			}
+			$res = $this->getByTitleLikeQuery($title4, $type, $source);
+			if (isset($res['id'])) {
+				return $res['id'];
+			}
+		}
+
+		// If there was not an exact title match, look for title with missing chars
+		// example release name :Zorro 1990, tvrage name Zorro (1990)
+		// Only search if the title contains more than one word to prevent incorrect matches
+		$pieces = explode(' ', $title);
+		if (count($pieces) > 1) {
+			$title4 = '%';
+			foreach ($pieces as $piece) {
+				$title4 .= str_replace(["'", "!"], "", $piece) . '%';
+			}
+			$res = $this->getByTitleLikeQuery($title4, $type, $source);
+			if (isset($res['id'])) {
+				return $res['id'];
+			}
+		}
+		return false;
+	}
+
+	/**
+	 * Supplementary function for getByTitle that queries for exact match
+	 *
+	 * @param        $title
+	 * @param        $type
+	 * @param int    $source
+	 *
+	 * @return array|bool
+	 */
+	public function getByTitleQuery($title, $type, $source = 0)
+	{
+		$return = false;
+		if ($title) {
+			$return = $this->pdo->queryOneRow(
+				sprintf("
+					SELECT v.id
+					FROM videos v
+					LEFT JOIN videos_akas va ON v.id = va.videos_id
+					WHERE (v.title = %1\$s OR va.title = %1\$s)
+					AND v.type = %2\$d %3\$s",
+					$this->pdo->escapeString($title),
+					$type,
+					($source > 0 ? 'AND v.source = ' . $source : '')
+				)
+			);
+		}
+		return $return;
+	}
+
+	/**
+	 * Supplementary function for getByTitle that queries for a like match
+	 *
+	 * @param        $title
+	 * @param        $type
+	 * @param int    $source
+	 *
+	 * @return array|bool
+	 */
+	public function getByTitleLikeQuery($title, $type, $source = 0)
+	{
+		$return = false;
+
+		if ($title) {
+			$return = $this->pdo->queryOneRow(
+				sprintf("
+					SELECT v.id
+					FROM videos v
+					LEFT JOIN videos_akas va ON v.id = va.videos_id
+					WHERE (v.title %1\$s
+					OR va.title %1\$s)
+					AND type = %2\$d %3\$s",
+					$this->pdo->likeString(rtrim($title, '%'), false, false),
+					$type,
+					($source > 0 ? 'AND v.source = ' . $source : '')
+				)
+			);
+		}
+		return $return;
+	}
+
+	/**
+	 * Inserts aliases for videos
+	 *
+	 * @param       $videoId
+	 * @param array $aliasArr
+	 */
+	public function addAliases($videoId, $aliasArr = array())
```

Comment:
just call the array aliases and type it so anything else will produce an error we can track:
public function addAliases($videoId, array $aliases = [])